quinn-proto/src/connection/bytes_source.rs

```diff
@@ -0,0 +1,249 @@
+//! Defines types which can be lazily converted into `Bytes` chunks
+
+use bytes::Bytes;
+
+/// A source of one or more buffers which can be converted into `Bytes` buffers on demand.
```

Comment:
By convention, this trait definition should go at the bottom of the module. Also, please drop the periods from the first sentence of each comment.

Reply:
The trait definition is still at the top, and the `Written` definition should go beneath it.

Reply:
As I wrote in the first comment in this thread, the trait definition should go at the bottom of the module.

Reply:
Sorry, I think I misunderstood. I put it beyond `Written` since I thought it would still be more important to most readers than the implementations, which are used only in a tiny set of places.

Reply:
I can move it down

Reply:
Actually I'm confused now. Either we move all dependencies down (`Written` + `BytesSource`), and keep the less important internal implementations on the top. Or we keep it as currently is. But I'm not sure if you want
- `Written` (publically exported)
- `BytesArray`
- `ByteSlice`
- `BytesSource`

?

Can you specify which order you intent?